backend/common/server/config.py

```diff
@@ -71,8 +73,12 @@ def apis_landing_page() -> str:
         """
 
     def before_request():
-        g.start = time.time()
+        g.start_time = time.time()
+        g.memory_usage = []
         g.request_id = generate_request_id()
+        g.track_memory = True
+        g.memory_thread = threading.Thread(target=_track_memory_usage)
```

Comment:
Does this measure the memory usage of the threads or the processes. If it is measuring the thread memory is it measuring `g.memory_thread` or the memory usage of the request.

Reply:
`memory_usage(-1,...)` tracks the memory usage of the current process. Threads (request thread and memory tracking thread) share the same process.

Reply:
the process has multiple threads running, so wont this value be the sum of all of the thread. So if multiple requests were being handles the memory usage would be for both threads.

Reply:
Each gunicorn worker is on its own process, but they are asynchronous gevent workers so it's true that memory usage will include other asynchronous requests happening at the same time. I don't think there's a way around this though